File: 0-easy/age-distribution/main.cpp
#include <iostream>
#include <sstream>
#include <fstream>

int main(int argc, char** argv)
{
    std::ifstream file(argv[1]);
    std::string line;

    while (std::getline(file, line))
    {
        int age;   
        std::stringstream ss(line);
        ss >> age;

        // Print line depending on the age.
        if (age < 0 || age > 100) {
            std::cout << "This program is for humans" << std::endl;
            continue;
        }

        if (age <= 2) {
            std::cout << "Still in Mama's arms" << std::endl;
        } else if (age <= 4) {
            std::cout << "Preschool Maniac" << std::endl;
        } else if (age <= 11) {
            std::cout << "Elementary school" << std::endl;
        } else if (age <= 14) {
            std::cout << "Middle school" << std::endl;
        } else if (age <= 18) {
            std::cout << "High school" << std::endl;
        } else if (age <= 22) {
            std::cout << "College" << std::endl;
        } else if (age <= 65) {
            std::cout << "Working for the man" << std::endl;
        } else {
            std::cout << "The Golden Years" << std::endl;
        }
    }
}

File: 0-easy/age-distribution/main.py
import sys

test_cases = open(sys.argv[1], 'r')
for test in test_cases:
    test = test.strip()
    if len(test) == 0:
        continue

    age = int(test)

    if age < 0 or age > 100:
        print "This program is for humans"
        continue

    if age <= 2:
        print "Still in Mama's arms"
    elif age <= 4:
        print "Preschool Maniac"
    elif age <= 11:
        print "Elementary school"
    elif age <= 14:
        print "Middle school"
    elif age <= 18:
        print "High school"
    elif age <= 22:
        print "College"
    elif age <= 65:
        print "Working for the man"
    else:
        print "The Golden Years"

test_cases.close()